File: app/src/main/java/dev/olog/basil/app/AppModule.java
package dev.olog.basil.app;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import dev.olog.basil.dagger.qualifier.ApplicationContext;

@Module
class AppModule {

    private final App app;

    AppModule(App app) {
        this.app = app;
    }

    @Provides
    @ApplicationContext
    Context provideContext(){
        return app;
    }

}

File: app/src/main/java/dev/olog/basil/AppModule.kt
package dev.olog.basil

import android.content.Context
import dagger.Binds
import dagger.Module
import dev.olog.basil.shared.dagger.qualifier.ApplicationContext

@Module
internal abstract class AppModule {

    @Binds
    @ApplicationContext
    internal abstract fun provideContext(instance: App): Context

}